platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,493 @@
+
+A `Genesis Roster` is an optional Roster that DevOps may provide to either start a Genesis Network process or Transplant
+an existing network. It is essential to distinguish the Genesis Roster from the Candidate or Active rosters.
+The Genesis Roster is a special roster used for the sole-purpose of bootstrapping a network. However, its structure is
+exactly the same as the Active or Candidate Rosters.
+The equivalent of the Genesis Roster in the current network is the `config.txt` file.
+DevOps and Services may choose to create the Genesis Roster from the `config.txt` file or some other mechanism.
+
+A new method will be added to the `PlatformBuilder` that will be used by the App to set the Genesis Roster.
+In most cases, this Genesis Roster will be null.
+
+```java
+//in PlatformBuilder
+
+/*
+  Set the Genesis Roster for the network.
+  <p>
+  This method is used to set the optional Genesis Roster for the network.
+  The Genesis Roster is a special roster used for the sole-purpose of bootstrapping a network.
+  The Genesis Roster, if set, is immutable and will be used to start a Genesis network Process or Network Transplant process.
+  <p>
+  @param genesisRoster the Genesis Roster for the network.
+ */
+void withGenesisRoster(@NonNull final Roster genesisRoster);
+```
+
+The logic to determine whether to invoke this method could be determined by a config flag,
+but this is yet to be implemented.
+
+#### New Transplant Procedure
+The App will decide a network transplant sequence based on the following heuristics:
+
+* Network Transplant process == Genesis Roster AND State provided
+  When both a Genesis Roster and State are provided, it signifies a network transplant.
+  The node will discard any existing candidate roster, rotate the active roster to the previous roster,
+  and adopt the provided Genesis Roster as the new Active Roster. This way, the provided network state and Genesis
+  Roster are adopted.
+  There will be new code required to create the Genesis Roster and pass it to the platform via the `PlatformBuilder` if
+  the network is intended to be in a Transplant mode.
+  The Services team will be responsible for implementing this, althouth the details are yet to be defined.
+
+* Genesis Network process == Genesis Roster, No State provided
+  When a Genesis Roster is provided, but no pre-existing state exists, it will indicate the creation of a new network.
+  The node will initiate the TSS key generation process (out of scope for this proposal),
+  create a genesis state with the provided Genesis Roster, and start participating in consensus from round 0.
+
+* Keep Network Settings (Normal restart) == No Genesis Roster AND State provided
+  If only a state is provided without a roster, the node will retain its existing network settings, including the active
+  roster.
+  This will be the typical behavior for a node restarting within an established network.
+  ![](Proposed%20Network%20Transplant%20flow.drawio.svg)
```

Comment:
I like this diagram, but a few things need to be fixed:
1. Add a start and end circle so that all flows end at the end circle.
2. There are three "yes" outputs from the "Genesis Roster Exist?" diamond. There must only be 1

Reply:
fixed